featurebyte/models/offline_store_feature_table.py

```diff
@@ -43,8 +48,119 @@ class OfflineStoreFeatureTableModel(FeatureByteCatalogBaseDocumentModel):
     feature_cluster: FeatureCluster
     output_column_names: List[str]
     output_dtypes: List[DBVarType]
-    entity_universe: EntityUniverseModel
+    internal_entity_universe: Optional[Dict[str, Any]] = Field(alias="entity_universe")
     entity_lookup_info: Optional[EntityRelationshipInfo]
+    feature_store_id: Optional[PydanticObjectId] = Field(default=None)
+
+    @root_validator
+    @classmethod
+    def _set_feature_store_id(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Set feature_store_id
+
+        Parameters
+        ----------
+        values : Dict[str, Any]
+            Values
+
+        Returns
+        -------
+        Dict[str, Any]
+        """
+        if not values.get("feature_store_id", None) and values.get("feature_cluster"):
+            values["feature_store_id"] = values["feature_cluster"].feature_store_id
+        return values
+
+    @property
+    def table_signature(self) -> Dict[str, Any]:
+        """
+        Get table signature
+
+        Returns
+        -------
+        Dict[str, Any]
+        """
+        entity_lookup_info = None
+        if self.entity_lookup_info:
+            entity_lookup_info = self.entity_lookup_info.dict(by_alias=True)
+        return {
+            "catalog_id": self.catalog_id,
+            "primary_entity_ids": self.primary_entity_ids,
+            "serving_names": self.serving_names,
+            "feature_job_setting": self.feature_job_setting.dict()
+            if self.feature_job_setting
+            else None,
+            "has_ttl": self.has_ttl,
+            "entity_lookup_info": entity_lookup_info,
+        }
+
+    @property
+    def entity_universe(self) -> EntityUniverseModel:
+        """
+        Get entity universe
+
+        Returns
+        -------
+        EntityUniverseModel
+
+        Raises
+        ------
+        ValueError
+            If entity_universe is not set
+        """
+        if self.internal_entity_universe is None:
+            raise ValueError("entity_universe is not set")
+        return EntityUniverseModel(**self.internal_entity_universe)
+
+    def _get_basename(self) -> str:
+        # max length of feature table name is 64
+        # reserving 7 characters for prefix (catalog name, `cat<num>_`, num is 1-999)
```

Comment:
Looking at the generated table names, I wonder if it would be clearer to spell out `cat` as `catalog`.